### PR TITLE
Disconnect sbrdsession

### PR DESCRIPTION
**Pull request by steevaavoo**
Finished splitting the function into 2 distinct, logical functions. Testing complete.

### CODE REVIEW

File: ManageRDSessions/ManageRDSessions.psm1 [hunk 1]
```diff
@@ -70,7 +66,7 @@ function Remove-sbRDSession {
     Process {
         # ForEach needed in order to process an array of objects passed in as a variable, as opposed to the Pipeline which feeds objects individually
         foreach ( $session in $RDSession ) {
-            if ($PSCmdlet.ShouldProcess("RDUser: [$($session.Username)] with Session ID [$($session.UnifiedSessionID)]", "[$Action]")) {
+            if ($PSCmdlet.ShouldProcess("RDUser: [$($session.Username)] with Session ID [$($session.UnifiedSessionID)]", "[Disconnect]")) {
```
Comment (adamrushuk) [inline ManageRDSessions/ManageRDSessions.psm1]: `[Disconnect]` should just be `Disconnect` as its not a variable; just text.

File: ManageRDSessions/ManageRDSessions.psm1 [hunk 2]
```diff
@@ -79,21 +75,51 @@ function Remove-sbRDSession {
...
-                    "LogOff" {
-                        Invoke-RDUserLogoff @params
-                        Write-Host "User [$($session.Username)] [Logged Off] from [$($session.HostServer)]"
-                    }
-                    "Disconnect" {
-                        Disconnect-RDUser @params
-                        Write-Host "User [$($session.Username)] [Disconnected] from [$($session.HostServer)]"
-                    }
-                    Default { }
+                Write-Verbose "Attempting [Disconnect] of $($session.Username) on $($session.HostServer)"
```
Comment (adamrushuk) [inline ManageRDSessions/ManageRDSessions.psm1]: [Disconnect] should just be Disconnect as its not a variable; just text.

File: ManageRDSessions/ManageRDSessions.psm1 [hunk 3]
```diff
@@ -79,21 +75,51 @@ function Remove-sbRDSession {
...
+                Disconnect-RDUser @params
+                Write-Host "User [$($session.Username)] [Disconnected] from [$($session.HostServer)]"
```
Comment (adamrushuk) [inline ManageRDSessions/ManageRDSessions.psm1]: [Disconnected] should just be Disconnected as its not a variable; just text.

File: ManageRDSessions/ManageRDSessions.psm1 [hunk 4]
```diff
@@ -79,21 +75,51 @@ function Remove-sbRDSession {
...
+    )
+
+    Begin {
+        # Intentionally empty
+    }
+    # Process block is required when receiving objects from the pipeline, otherwise only last object will be received
+    Process {
+        # ForEach needed in order to process an array of objects passed in as a variable, as opposed to the Pipeline which feeds objects individually
+        foreach ( $session in $RDSession ) {
+            if ($PSCmdlet.ShouldProcess("RDUser: [$($session.Username)] with Session ID [$($session.UnifiedSessionID)]", "[LogOff]")) {
```
Comment (adamrushuk) [inline ManageRDSessions/ManageRDSessions.psm1]: [LogOff] should just be LogOff as its not a variable; just text.

File: ManageRDSessions/ManageRDSessions.psm1 [hunk 5]
```diff
@@ -79,21 +75,51 @@ function Remove-sbRDSession {
...
 
+                $params = @{
+                    'HostServer'       = $session.HostServer
+                    'UnifiedSessionID' = $session.UnifiedSessionID
+                    'ErrorAction'      = 'Stop'
+                    'Force'            = $true
                 }
 
+                Write-Verbose "Attempting [LogOff] of $($session.Username) on $($session.HostServer)"
```
Comment (adamrushuk) [inline ManageRDSessions/ManageRDSessions.psm1]: [LogOff] should just be LogOff as its not a variable; just text.

File: ManageRDSessions/ManageRDSessions.psm1 [hunk 6]
```diff
@@ -79,21 +75,51 @@ function Remove-sbRDSession {
...
+                Invoke-RDUserLogoff @params
+                Write-Host "User [$($session.Username)] [Logged Off] from [$($session.HostServer)]"
```
Comment (adamrushuk) [inline ManageRDSessions/ManageRDSessions.psm1]: You know!

File: vagrant/provision/rds01/Configure-RDS.ps1 [hunk 7]
```diff
@@ -0,0 +1,36 @@
+# Parameters Hard Coded here because this needs to be run inside host.
+param(
+    $RDHostName = "rds01.lab.milliondollar.me.uk",
+    $RDCollectionName = "MillionDollarCollection"
```
Comment (adamrushuk) [inline vagrant/provision/rds01/Configure-RDS.ps1:4]: Both of these values can be passed in from your Vagrantfile.
See how I've done this in the other scripts, eg:
`machine.vm.provision "shell", path: "Vagrant/provision/all/Join-Domain.ps1", args: [domain_name, domain_admin_un, domain_admin_pw, dc01_ip]`

For each variable you need to pass in, you will need to create a new var at the top of Vagrantfile.
Nothing wrong with having defaults though :)

**Review by adamrushuk — CHANGES_REQUESTED** [review]
In general all is looking good.
The next level will be proper error handling; I also missed doing this in my original Vagrant scripts. Sorry!

File: vagrant/provision/rds01/Configure-RDS.ps1 (hunk 7 above)
Comment (steevaavoo) [reply vagrant/provision/rds01/Configure-RDS.ps1:4]: Ah yes - I thought you'd pick up on this. This script is actually not useable through Vagrant  provisioning thanks to the double-hop WinRM issue. So I've just hard-coded those variables as I'll need to call the script from within the VM anyway. I had originally passed them in, until I hit this issue.

Comment (steevaavoo) [reply vagrant/provision/rds01/Configure-RDS.ps1:4]: For clarity - RD tries to use WinRM on the local machine you're trying to set it up on - which spins the whole remote provisioning out.

File: ManageRDSessions/ManageRDSessions.psm1 (hunk 6 above)
Comment (steevaavoo) [reply ManageRDSessions/ManageRDSessions.psm1]: Efficiency: 5x done ;)

File: vagrant/provision/rds01/Configure-RDS.ps1 (hunk 7 above)
Comment (adamrushuk) [reply vagrant/provision/rds01/Configure-RDS.ps1:4]: Ah ok. Something to workout in a future version perhaps

**Comment by adamrushuk** [comment]
LGTM!

**Review by adamrushuk — APPROVED** [review]

**Pull request merged**